event/gov_check_votes.go

```diff
@@ -0,0 +1,93 @@
+package event
+
+import (
+	"github.com/Oneledger/protocol/data/governance"
+	"github.com/Oneledger/protocol/data/jobs"
+	"github.com/Oneledger/protocol/storage"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+var _ jobs.Job = &JobGovCheckVotes{}
+
+type JobGovCheckVotes struct {
+	ProposalID governance.ProposalID
+	JobID      string
+	RetryCount int
+	Status     jobs.Status
+}
+
+func NewGovCheckVotesJob(proposalID governance.ProposalID, status governance.ProposalStatus) *JobGovCheckVotes {
+	return &JobGovCheckVotes{
+		ProposalID: proposalID,
+		JobID:      proposalID.String() + storage.DB_PREFIX + strconv.Itoa(int(status)),
+		RetryCount: 0,
+		Status:     0,
+	}
+}
+
+func (j *JobGovCheckVotes) DoMyJob(ctx interface{}) {
+	govCtx, _ := ctx.(*JobsContext)
+	proposalMaster := govCtx.ProposalMaster
+
+	//Get Proposal
+	proposal, err := proposalMaster.Proposal.WithPrefixType(governance.ProposalStateActive).Get(j.ProposalID)
+	if err != nil {
+		j.Status = jobs.Failed
+		govCtx.Logger.Error("gov_check_votes: proposal not found in active prefix")
+		return
+	}
+
+	//Check if proposal is in voting state
+	if proposal.Status != governance.ProposalStatusVoting {
+		j.Status = jobs.Failed
```

Comment:
its already checked before creating the job proposal.Status == governance.ProposalStatusVoting  , is this required here ?